tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -937,31 +975,86 @@ def _generate_build_extra_metadata_for_tests(tests, bazel_rules):
     'deps("//src/proto/...")',
 ]
 
+# Step 1: run a bunch of "bazel query --output xml" queries to collect
+# the raw build metadata from the bazel build.
+# At the end of this step we will have a dictionary of bazel rules
+# that are interesting to us (libraries, binaries, etc.) along
+# with their most important metadata (sources, headers, dependencies)
 bazel_rules = {}
 for query in _BAZEL_DEPS_QUERIES:
     bazel_rules.update(
         _extract_rules_from_bazel_xml(_bazel_query_xml_tree(query)))
 
+# Step 1a: Knowing the transitive closure of dependencies will make
+# the postprocessing simpler, so compute the info for all our rules.
```

Comment:
I assume this step modifies `bazel_rules` in place.  Can you show an example of how a given rule might be modified, so a reader can tell what the data should look like at this point?